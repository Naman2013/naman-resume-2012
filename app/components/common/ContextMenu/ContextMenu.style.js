import css from 'styled-jsx/css';
import { primaryFont } from 'styles/variables/fonts';
import { faintGray, astronaut } from 'styles/variables/colors_tiles_v4';

export const CONTAINER_WIDTH = 440;

export default css`
  .root {
    width: 100px;
    overflow: hidden;
  }

  .menu-container {
    position: absolute;
    padding: 0;
    margin: 0;
    top: 110px;
    width: ${Math.abs(CONTAINER_WIDTH)}px;
    height: 100%;
    background-color: ${faintGray};
    transition: right ease-out 0.15s;
  }

  .header-container {
    margin: 0;
    padding: 40px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    color: ${astronaut};
    background-color: white;
  }

  .context-header {
    margin: 0;
    padding: 0;
    font-family: ${primaryFont};
    font-size: 12px;
    text-transform: uppercase;
    letter-spacing: 1.5px;
    font-weight: 800;
  }

  .available-sections {
    margin: 0;
    padding: 0;
    font-family: ${primaryFont};
    font-size: 12px;
  }
`;
