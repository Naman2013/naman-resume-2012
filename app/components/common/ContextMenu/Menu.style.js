import css from 'styled-jsx/css';

export default css`
.menu-list {
  margin: 0;
  padding: 0;
  padding-top: 10px;
}
`;
