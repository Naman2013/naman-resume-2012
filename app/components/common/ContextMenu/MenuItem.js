import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import MiraDot from 'atoms/icons/MiraDot';
import style from './MenuItem.style';

const MenuItem = ({ title, linkURL, iconUrl }) => (
  <Fragment>
    <li className="menu-item">
      <span className="item-container">
        <Link activeClassName="active" to={linkURL}>
          <span className="dot-container">
            <MiraDot />
          </span>
          {title}
        </Link>
        {iconUrl && (
          <div className="iconContainer">
            <img className="icon" src={iconUrl} />
          </div>
        )}
      </span>
    </li>
    <style jsx>{style}</style>
  </Fragment>
);

MenuItem.propTypes = {
  title: PropTypes.string.isRequired,
  linkURL: PropTypes.string.isRequired,
  iconUrl: PropTypes.string,
};

MenuItem.defaultProps = {
  iconUrl: null,
};

export default MenuItem;
