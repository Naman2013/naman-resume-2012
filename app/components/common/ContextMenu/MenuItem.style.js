import css from 'styled-jsx/css';
import {
  astronaut,
  geyser,
  nightfall,
} from 'app/styles/variables/colors_tiles_v4';
import { secondaryFont } from 'app/styles/variables/fonts';

export default css`
  .menu-item {
    position: relative;
    margin: 0;
    padding: 0;
    margin-left: 70px;
    width: 75%;
    color: ${astronaut};
  }

  .dot-container {
    position: absolute;
    left: -50px;
    top: 43px;
  }

  .item-container {
    display: flex;
    justify-content: space-between;
    border-bottom: 1px solid ${geyser};
  }

  .iconContainer {
    width: 50px;
    height: 50px;
    background-color: ${nightfall};
    border-radius: 50%;
    margin-top: 20px;
  }

  .icon {
    width: 34px;
    height: 34px;
    position: relative;
    top: 8px;
    left: 8px;
  }

  .menu-item :global(a) {
    display: block;
    padding-bottom: 30px;
    padding-top: 40px;
    margin-left: -15px;
    padding-left: 20px;
    font-family: ${secondaryFont};
    text-transform: capitalize;
    text-decoration: none;
    color: ${astronaut};
  }

  .menu-item :global(a .dot-container) {
    opacity: 0;
  }

  .menu-item :global(a.active .dot-container) {
    opacity: 1;
  }
`;
