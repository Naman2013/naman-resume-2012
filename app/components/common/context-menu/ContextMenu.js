import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { white } from '../../../styles/variables/colors';

const {
  bool,
  number,
  string,
  func,
} = PropTypes;

class ContextMenu extends Component {

  static propTypes = {
    menuHeight: number,
    menuWidth: number,
    distanceFromTrigger: number,
    backgroundColor: string,
    onShow: func,
  }

  static defaultProps = {
    // how tall the menu will be (in pixels)
    menuHeight: 300,
    // how wide the menu will be (in pixels)
    menuWidth: 200,
    // padding from the trigger button
    distanceFromTrigger: 15,
    // background color for menu
    backgroundColor: white,
    onShow: null,
  };

  state = {
    triggerElement: null,
    left: -(this.props.menuWidth + this.props.distanceFromTrigger),
    top: -(this.props.menuHeight / 2),
    showMenu: false,
  }

  handleContextClick = (e) => {
    const { onShow } = this.props;
    const { showMenu } = this.state;

    if (!showMenu) {
      onShow();
    }

    this.setState({
      showMenu: !showMenu,
    });
  }

  hideMenu = (e) => {
    this.setState({
      showMenu: false,
    });
  }

  render() {
    const {
      menuHeight,
      menuWidth,
      backgroundColor,
      children,
      className,
    } = this.props;

    const { left, top, showMenu } = this.state;

    const menuRootStyle = {
      position: 'absolute',
      top: `${top}px`,
      left: `${left}px`,
      height: `${menuHeight}px`,
      width: `${menuWidth}px`,
      backgroundColor,
    };

    const containerClass = classnames({
      notShown: !showMenu,
    });

    const rootClasses = classnames({
      [className]: !!className,
    });

    return (
      <div
        className={containerClass}
      >
        <div
          className="arrow-right"
        />
        <div
          onMouseLeave={this.hideMenu}
          style={menuRootStyle}
          className={rootClasses}
        >
          {children}
        </div>
        <style jsx>{`

          .notShown {
            visibility: hidden;
          }

          .arrow-right {
            position: absolute;
            left: 15px;
            width: 0;
            height: 0;
            border-top: 20px solid transparent;
            border-bottom: 20px solid transparent;
            border-left: 20px solid ${white};
          }

        `}</style>
      </div>
    );
  }
}

export default ContextMenu;
