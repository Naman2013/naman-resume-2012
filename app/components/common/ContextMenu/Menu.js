import React from 'react';
import PropTypes from 'prop-types';
import MenuItem from './MenuItem';
import style from './Menu.style';

const Menu = ({ list }) => (
  <ol className="menu-list">
    {list.map(item => (<MenuItem key={`context-menu-${item.title}`} {...item} />))}
    <style jsx>{style}</style>
  </ol>
);

Menu.propTypes = {
  list: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    linkURL: PropTypes.string.isRequired,
  })).isRequired,
};

export default Menu;
