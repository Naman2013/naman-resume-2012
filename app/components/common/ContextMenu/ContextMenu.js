import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Menu from './Menu';
import Fade from 'components/common/Fade';
import VanillaButton from 'atoms/VanillaButton';
import Dots from 'atoms/icons/Dots';
import Close from 'atoms/icons/Close';
import { thatBlue, astronaut } from 'styles/variables/colors_tiles_v4';
import style, { CONTAINER_WIDTH } from './ContextMenu.style';

const OPEN_LOCATION = 0;
export const CLOSE_LOCATION = -CONTAINER_WIDTH;

class ContextMenu extends Component {
  static propTypes = {
    isOpen: PropTypes.bool,
    menuTopAdjustment: PropTypes.number,
    title: PropTypes.string.isRequired,
    count: PropTypes.number.isRequired,
    list: PropTypes.arrayOf(PropTypes.shape({
      title: PropTypes.string,
      linkURL: PropTypes.string,
    })).isRequired,
  };

  static defaultProps = {
    isOpen: false,
    menuTopAdjustment: 98,
  };

  state = {
    isOpen: this.props.isOpen,
  };

  toggle = () => { this.setState(prevState => ({ isOpen: !prevState.isOpen })); }

  render() {
    const { isOpen } = this.state;
    const containerLeft = (isOpen) ? OPEN_LOCATION : CLOSE_LOCATION;

    return (
      <div className="root">
        <Fade isHidden={!isOpen}>
          <div className="application-veil" />
        </Fade>

        <VanillaButton
          handleClick={this.toggle}
          theme={{
            width: '100px',
            height: '100px',
            backgroundColor: (isOpen) ? 'white' : thatBlue,
            position: 'relative',
            zIndex: '10',
          }}
        >
          {(isOpen)
            ? <Close theme={{ fillColor: astronaut }} />
            : <div className="dots-container"><Dots /></div>}
        </VanillaButton>

        <div
          className="menu-container"
          style={{
            right: `${containerLeft}px`,
          }}
        >
          <div className="header-container">
            <h5 className="context-header">{this.props.title}</h5>
            <p className="available-sections">( {this.props.count} )</p>
          </div>

          <Menu list={this.props.list} />
        </div>

        <style jsx>{style}</style>
        <style jsx>
          {`
            .menu-container {
              top: ${this.props.menuTopAdjustment}px;
            }

            .application-veil {
              width: 100vw;
              height: 100%;
              position: absolute;
              top: ${this.props.menuTopAdjustment}px;
              left: 0;
              background-color: rgba(0, 0, 0, 0.5);
              pointer-events: none;
            }
          `}
        </style>
      </div>
    );
  }
}

export default ContextMenu;
