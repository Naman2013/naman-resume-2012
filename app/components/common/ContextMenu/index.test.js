import React from 'react';
import { shallow } from 'enzyme';
import ContextMenu from './ContextMenu';
import Menu from './Menu';
import MenuItem from './MenuItem';

describe('ContextMenu', () => {
  const shallowWrapper = shallow(<ContextMenu />);
  it('should render correctly', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });

  describe('clicking the menu button', () => {
    it('initializes with `isOpen` to `false`', () => {
      expect(shallowWrapper.state('isOpen')).toEqual(false);
    });

    it('first click of menu button will toggle `isOpen` to `true`', () => {
      shallowWrapper.find('VanillaButton').dive().find('button').simulate('click');
      expect(shallowWrapper.state('isOpen')).toEqual(true);
    });

    it('second click of menu will toggle `isOpen` back to `false`', () => {
      shallowWrapper.find('VanillaButton').dive().find('button').simulate('click');
      expect(shallowWrapper.state('isOpen')).toEqual(false);
    });
  });
});

describe('Menu', () => {
  const menuShallowWrapper = shallow(<Menu />);
  it('should render correctly', () => {
    expect(menuShallowWrapper).toMatchSnapshot();
  });
});

describe('MenuItem', () => {
  const menuItemShallowWrapper = shallow(<MenuItem />);
  it('should render correctly', () => {
    expect(menuItemShallowWrapper).toMatchSnapshot();
  });
});
