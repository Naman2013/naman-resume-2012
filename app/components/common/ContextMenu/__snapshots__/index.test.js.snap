// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextMenu should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContextMenu
    isOpen={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <VanillaButton
          handleClick={[Function]}
          theme={
            Object {
              "backgroundColor": "#253446",
              "height": "100px",
              "width": "100px",
            }
          }
        >
          <Dots
            height={4}
            theme={
              Object {
                "circleColor": "white",
              }
            }
            width={18}
          />
        </VanillaButton>,
        <div
          className="jsx-2499475602 menu-container"
          style={
            Object {
              "right": "-440px",
            }
          }
        >
          <div
            className="jsx-2499475602 header-container"
          >
            <h5
              className="jsx-2499475602 context-header"
            >
              Sample title
            </h5>
            <p
              className="jsx-2499475602 available-sections"
            >
              ( 8 )
            </p>
          </div>
          <Menu />
        </div>,
        <JSXStyle
          css={
            Array [
              ".root.jsx-2499475602{width:100px;overflow:hidden;}",
              ".menu-container.jsx-2499475602{position:absolute;padding:0;margin:0;top:110px;width:440px;height:100%;background-color:#f8f8f8;-webkit-transition:right ease-out 0.15s;transition:right ease-out 0.15s;}",
              ".header-container.jsx-2499475602{margin:0;padding:40px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;color:#415671;background-color:white;}",
              ".context-header.jsx-2499475602{margin:0;padding:0;font-family:brandon-grotesque, sans-serif;font-size:12px;text-transform:uppercase;-webkit-letter-spacing:1.5px;-moz-letter-spacing:1.5px;-ms-letter-spacing:1.5px;letter-spacing:1.5px;font-weight:800;}",
              ".available-sections.jsx-2499475602{margin:0;padding:0;font-family:brandon-grotesque, sans-serif;font-size:12px;}",
            ]
          }
          styleId="2499475602"
        />,
      ],
      "className": "jsx-2499475602 root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Dots
            height={4}
            theme={
              Object {
                "circleColor": "white",
              }
            }
            width={18}
          />,
          "handleClick": [Function],
          "theme": Object {
            "backgroundColor": "#253446",
            "height": "100px",
            "width": "100px",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "height": 4,
            "theme": Object {
              "circleColor": "white",
            },
            "width": 18,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="jsx-2499475602 header-container"
            >
              <h5
                className="jsx-2499475602 context-header"
              >
                Sample title
              </h5>
              <p
                className="jsx-2499475602 available-sections"
              >
                ( 8 )
              </p>
            </div>,
            <Menu />,
          ],
          "className": "jsx-2499475602 menu-container",
          "style": Object {
            "right": "-440px",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h5
                  className="jsx-2499475602 context-header"
                >
                  Sample title
                </h5>,
                <p
                  className="jsx-2499475602 available-sections"
                >
                  ( 8 )
                </p>,
              ],
              "className": "jsx-2499475602 header-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sample title",
                  "className": "jsx-2499475602 context-header",
                },
                "ref": null,
                "rendered": "Sample title",
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "( 8 )",
                  "className": "jsx-2499475602 available-sections",
                },
                "ref": null,
                "rendered": "( 8 )",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            ".root.jsx-2499475602{width:100px;overflow:hidden;}",
            ".menu-container.jsx-2499475602{position:absolute;padding:0;margin:0;top:110px;width:440px;height:100%;background-color:#f8f8f8;-webkit-transition:right ease-out 0.15s;transition:right ease-out 0.15s;}",
            ".header-container.jsx-2499475602{margin:0;padding:40px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;color:#415671;background-color:white;}",
            ".context-header.jsx-2499475602{margin:0;padding:0;font-family:brandon-grotesque, sans-serif;font-size:12px;text-transform:uppercase;-webkit-letter-spacing:1.5px;-moz-letter-spacing:1.5px;-ms-letter-spacing:1.5px;letter-spacing:1.5px;font-weight:800;}",
            ".available-sections.jsx-2499475602{margin:0;padding:0;font-family:brandon-grotesque, sans-serif;font-size:12px;}",
          ],
          "styleId": "2499475602",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <VanillaButton
            handleClick={[Function]}
            theme={
              Object {
                "backgroundColor": "#253446",
                "height": "100px",
                "width": "100px",
              }
            }
          >
            <Dots
              height={4}
              theme={
                Object {
                  "circleColor": "white",
                }
              }
              width={18}
            />
          </VanillaButton>,
          <div
            className="jsx-2499475602 menu-container"
            style={
              Object {
                "right": "-440px",
              }
            }
          >
            <div
              className="jsx-2499475602 header-container"
            >
              <h5
                className="jsx-2499475602 context-header"
              >
                Sample title
              </h5>
              <p
                className="jsx-2499475602 available-sections"
              >
                ( 8 )
              </p>
            </div>
            <Menu />
          </div>,
          <JSXStyle
            css={
              Array [
                ".root.jsx-2499475602{width:100px;overflow:hidden;}",
                ".menu-container.jsx-2499475602{position:absolute;padding:0;margin:0;top:110px;width:440px;height:100%;background-color:#f8f8f8;-webkit-transition:right ease-out 0.15s;transition:right ease-out 0.15s;}",
                ".header-container.jsx-2499475602{margin:0;padding:40px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;color:#415671;background-color:white;}",
                ".context-header.jsx-2499475602{margin:0;padding:0;font-family:brandon-grotesque, sans-serif;font-size:12px;text-transform:uppercase;-webkit-letter-spacing:1.5px;-moz-letter-spacing:1.5px;-ms-letter-spacing:1.5px;letter-spacing:1.5px;font-weight:800;}",
                ".available-sections.jsx-2499475602{margin:0;padding:0;font-family:brandon-grotesque, sans-serif;font-size:12px;}",
              ]
            }
            styleId="2499475602"
          />,
        ],
        "className": "jsx-2499475602 root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Dots
              height={4}
              theme={
                Object {
                  "circleColor": "white",
                }
              }
              width={18}
            />,
            "handleClick": [Function],
            "theme": Object {
              "backgroundColor": "#253446",
              "height": "100px",
              "width": "100px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "height": 4,
              "theme": Object {
                "circleColor": "white",
              },
              "width": 18,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="jsx-2499475602 header-container"
              >
                <h5
                  className="jsx-2499475602 context-header"
                >
                  Sample title
                </h5>
                <p
                  className="jsx-2499475602 available-sections"
                >
                  ( 8 )
                </p>
              </div>,
              <Menu />,
            ],
            "className": "jsx-2499475602 menu-container",
            "style": Object {
              "right": "-440px",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h5
                    className="jsx-2499475602 context-header"
                  >
                    Sample title
                  </h5>,
                  <p
                    className="jsx-2499475602 available-sections"
                  >
                    ( 8 )
                  </p>,
                ],
                "className": "jsx-2499475602 header-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sample title",
                    "className": "jsx-2499475602 context-header",
                  },
                  "ref": null,
                  "rendered": "Sample title",
                  "type": "h5",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "( 8 )",
                    "className": "jsx-2499475602 available-sections",
                  },
                  "ref": null,
                  "rendered": "( 8 )",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              ".root.jsx-2499475602{width:100px;overflow:hidden;}",
              ".menu-container.jsx-2499475602{position:absolute;padding:0;margin:0;top:110px;width:440px;height:100%;background-color:#f8f8f8;-webkit-transition:right ease-out 0.15s;transition:right ease-out 0.15s;}",
              ".header-container.jsx-2499475602{margin:0;padding:40px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;color:#415671;background-color:white;}",
              ".context-header.jsx-2499475602{margin:0;padding:0;font-family:brandon-grotesque, sans-serif;font-size:12px;text-transform:uppercase;-webkit-letter-spacing:1.5px;-moz-letter-spacing:1.5px;-ms-letter-spacing:1.5px;letter-spacing:1.5px;font-weight:800;}",
              ".available-sections.jsx-2499475602{margin:0;padding:0;font-family:brandon-grotesque, sans-serif;font-size:12px;}",
            ],
            "styleId": "2499475602",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Menu should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <MenuItem />,
        <JSXStyle
          css={
            Array [
              ".menu-list.jsx-3731609075{margin:0;padding:0;padding-top:10px;}",
            ]
          }
          styleId="3731609075"
        />,
      ],
      "className": "jsx-3731609075 menu-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            ".menu-list.jsx-3731609075{margin:0;padding:0;padding-top:10px;}",
          ],
          "styleId": "3731609075",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "ol",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <MenuItem />,
          <JSXStyle
            css={
              Array [
                ".menu-list.jsx-3731609075{margin:0;padding:0;padding-top:10px;}",
              ]
            }
            styleId="3731609075"
          />,
        ],
        "className": "jsx-3731609075 menu-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              ".menu-list.jsx-3731609075{margin:0;padding:0;padding-top:10px;}",
            ],
            "styleId": "3731609075",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ol",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MenuItem should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuItem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <li
          className="jsx-3645762491 menu-item"
        >
          <span
            className="jsx-3645762491 dot-container"
          >
            <MiraDot
              theme={
                Object {
                  "dimension": 10,
                }
              }
            />
          </span>
          <a
            className="jsx-3645762491 action"
            href="#"
          >
            Astronomical Time
          </a>
        </li>,
        <JSXStyle
          css={
            Array [
              ".menu-item.jsx-3645762491{margin:0;padding:0;margin-left:70px;width:75%;color:#415671;}",
              ".action.jsx-3645762491{display:block;padding-bottom:30px;padding-top:40px;margin-left:-15px;padding-left:20px;font-family:adobe-garamond-pro, serif;border-bottom:1px solid #d9dee5;text-transform:capitalize;-webkit-text-decoration:none;text-decoration:none;color:#415671;}",
            ]
          }
          styleId="3645762491"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="jsx-3645762491 dot-container"
            >
              <MiraDot
                theme={
                  Object {
                    "dimension": 10,
                  }
                }
              />
            </span>,
            <a
              className="jsx-3645762491 action"
              href="#"
            >
              Astronomical Time
            </a>,
          ],
          "className": "jsx-3645762491 menu-item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <MiraDot
                theme={
                  Object {
                    "dimension": 10,
                  }
                }
              />,
              "className": "jsx-3645762491 dot-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "theme": Object {
                  "dimension": 10,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Astronomical Time",
              "className": "jsx-3645762491 action",
              "href": "#",
            },
            "ref": null,
            "rendered": "Astronomical Time",
            "type": "a",
          },
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            ".menu-item.jsx-3645762491{margin:0;padding:0;margin-left:70px;width:75%;color:#415671;}",
            ".action.jsx-3645762491{display:block;padding-bottom:30px;padding-top:40px;margin-left:-15px;padding-left:20px;font-family:adobe-garamond-pro, serif;border-bottom:1px solid #d9dee5;text-transform:capitalize;-webkit-text-decoration:none;text-decoration:none;color:#415671;}",
          ],
          "styleId": "3645762491",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <li
            className="jsx-3645762491 menu-item"
          >
            <span
              className="jsx-3645762491 dot-container"
            >
              <MiraDot
                theme={
                  Object {
                    "dimension": 10,
                  }
                }
              />
            </span>
            <a
              className="jsx-3645762491 action"
              href="#"
            >
              Astronomical Time
            </a>
          </li>,
          <JSXStyle
            css={
              Array [
                ".menu-item.jsx-3645762491{margin:0;padding:0;margin-left:70px;width:75%;color:#415671;}",
                ".action.jsx-3645762491{display:block;padding-bottom:30px;padding-top:40px;margin-left:-15px;padding-left:20px;font-family:adobe-garamond-pro, serif;border-bottom:1px solid #d9dee5;text-transform:capitalize;-webkit-text-decoration:none;text-decoration:none;color:#415671;}",
              ]
            }
            styleId="3645762491"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="jsx-3645762491 dot-container"
              >
                <MiraDot
                  theme={
                    Object {
                      "dimension": 10,
                    }
                  }
                />
              </span>,
              <a
                className="jsx-3645762491 action"
                href="#"
              >
                Astronomical Time
              </a>,
            ],
            "className": "jsx-3645762491 menu-item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <MiraDot
                  theme={
                    Object {
                      "dimension": 10,
                    }
                  }
                />,
                "className": "jsx-3645762491 dot-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "theme": Object {
                    "dimension": 10,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Astronomical Time",
                "className": "jsx-3645762491 action",
                "href": "#",
              },
              "ref": null,
              "rendered": "Astronomical Time",
              "type": "a",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              ".menu-item.jsx-3645762491{margin:0;padding:0;margin-left:70px;width:75%;color:#415671;}",
              ".action.jsx-3645762491{display:block;padding-bottom:30px;padding-top:40px;margin-left:-15px;padding-left:20px;font-family:adobe-garamond-pro, serif;border-bottom:1px solid #d9dee5;text-transform:capitalize;-webkit-text-decoration:none;text-decoration:none;color:#415671;}",
            ],
            "styleId": "3645762491",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
